vignettes/modification.Rmd

```diff
@@ -119,6 +126,29 @@ xml_add_child(t1, read_xml("<test/>"))
 x
 ```
 
+## Removing nodes ##
+The `xml_remove_node()` can be used to remove a node (and it's children) from a
+tree. The default behavior is to unlink the node from the tree, but does not
+free the memory for the node, so R objects pointing to the node are still
+valid. This allows code like the following to work without crashing R
+
+```r
+x <- read_xml("<foo><bar><baz/><bar></foo>")
+x1 <- x %>% xml_children() %>% .[[1]]
+x2 <- x1 %>% xml_children() %>% .[[1]]
+
+xml_remove(x1)
+rm(x1)
+gc()
+
+x2
+```
+
+Setting the argument `free = TRUE` when calling `xml_remove_node()`
+also frees the memory when the node is unlinked. In this case it is the
+responsibility of the user to remove all objects pointing to the node or its
+children, as they no longer point to valid memory.
```

Comment:
And I think you should include an example showing how you should use `free = TRUE`.  Maybe mention that `xml_remove()` should always be called with `free = TRUE`, but you need to be careful not to have multiple R objects that point to the same object?

This case is pretty pernicious:

```R
 <- read_xml("<a><b /><b><b /></b></a>")
xml_find_all(x, "//b")
```